diff -r 0eb354feeb31 make/lib/CoreLibraries.gmk
--- openjdk/jdk/make/lib/CoreLibraries.gmk	Sat Jun 25 23:29:26 2016 +0100
+++ openjdk/jdk/make/lib/CoreLibraries.gmk	Fri Sep 30 21:05:01 2016 +0100
@@ -142,7 +142,7 @@
 
 LIBJAVA_CFLAGS := $(foreach dir, $(LIBJAVA_SRC_DIRS), -I$(dir)) \
     -I$(JDK_TOPDIR)/src/share/native/java/lang/fdlibm/include \
-    -DARCHPROPNAME='"$(OPENJDK_TARGET_CPU_OSARCH)"'
+    -DARCHPROPNAME='"arm"'
 
 LIBJAVA_CFLAGS += -DJDK_MAJOR_VERSION='"$(JDK_MAJOR_VERSION)"' \
     -DJDK_MINOR_VERSION='"$(JDK_MINOR_VERSION)"' \
